### PR TITLE
feat(integrations): adds tests for integration code mappings

### PR DESCRIPTION
**Pull request by scefali**
This PR adds tests for the new integration code mappings UI that we are building. I also made the repo dropdown a non-legacy dropdown and added a prop to the field so this can be specified via `FieldFromConfig`.

### CODE REVIEW

File: tests/js/spec/views/organizationIntegrations/integrationCodeMappings.spec.jsx [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    ]);
+
+    wrapper = mountWithTheme(
+      <IntegrationCodeMappings organization={org} integration={integration} />
+    );
+  });
+  it('shows the paths', () => {
+    expect(wrapper.find('RepoName').length).toEqual(2);
+    expect(wrapper.find('RepoName').at(0).text()).toEqual('example/repo-name');
+    expect(wrapper.find('RepoName').at(1).text()).toEqual('example/hello-there');
```
Comment (priscilawebdev) [inline tests/js/spec/views/organizationIntegrations/integrationCodeMappings.spec.jsx:84]:
```suggestion
    expect(wrapper.find('RepoName').at(1).text()).toEqual(repos[1]. name);
```

File: tests/js/spec/views/organizationIntegrations/integrationCodeMappings.spec.jsx [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+      [`/organizations/${org.slug}/repos/`, repos],
+    ]);
+
+    wrapper = mountWithTheme(
+      <IntegrationCodeMappings organization={org} integration={integration} />
+    );
+  });
+  it('shows the paths', () => {
+    expect(wrapper.find('RepoName').length).toEqual(2);
+    expect(wrapper.find('RepoName').at(0).text()).toEqual('example/repo-name');
```
Comment (priscilawebdev) [inline tests/js/spec/views/organizationIntegrations/integrationCodeMappings.spec.jsx:83]:
```suggestion
    expect(wrapper.find('RepoName').at(0).text()).toEqual(repos[0].name);
```

File: tests/js/spec/views/organizationIntegrations/integrationCodeMappings.spec.jsx [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+    expect(wrapper.find('RepoName').at(1).text()).toEqual('example/hello-there');
+  });
+  it('opens modal', async () => {
+    expect(wrapper.find('input[name="stackRoot"]')).toHaveLength(0);
+    wrapper.find('button[aria-label="Add Mapping"]').first().simulate('click');
+    await tick();
+    expect(wrapper.find('input[name="stackRoot"]')).toHaveLength(1);
+  });
+  it('create new config', async () => {
```
Comment (priscilawebdev) [inline tests/js/spec/views/organizationIntegrations/integrationCodeMappings.spec.jsx:92]:
```suggestion
  });
  
  it('create new config', async () => {
```

File: tests/js/spec/views/organizationIntegrations/integrationCodeMappings.spec.jsx [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+    wrapper = mountWithTheme(
+      <IntegrationCodeMappings organization={org} integration={integration} />
+    );
+  });
+  it('shows the paths', () => {
+    expect(wrapper.find('RepoName').length).toEqual(2);
+    expect(wrapper.find('RepoName').at(0).text()).toEqual('example/repo-name');
+    expect(wrapper.find('RepoName').at(1).text()).toEqual('example/hello-there');
+  });
+  it('opens modal', async () => {
```
Comment (priscilawebdev) [inline tests/js/spec/views/organizationIntegrations/integrationCodeMappings.spec.jsx:86]:
```suggestion
  });
  
  it('opens modal', async () => {
```

File: tests/js/spec/views/organizationIntegrations/integrationCodeMappings.spec.jsx [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+          projectId: projects[1].id,
+          repositoryId: repos[1].id,
+          stackRoot: 'my/root',
+          sourceRoot: 'hey/dude',
+          defaultBranch: 'release',
+        },
+      })
+    );
+  });
+  it('edit existing config', async () => {
```
Comment (priscilawebdev) [inline tests/js/spec/views/organizationIntegrations/integrationCodeMappings.spec.jsx:132]:
```suggestion
  });
  
  it('edit existing config', async () => {
```

File: tests/js/spec/views/organizationIntegrations/integrationCodeMappings.spec.jsx [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+    const url = `/organizations/${org.slug}/integrations/${integration.id}/repo-project-path-configs/${pathConfig1.id}/`;
+    const editMock = Client.addMockResponse({
+      url,
+      method: 'PUT',
+      body: TestStubs.RepositoryProjectPathConfig(projects[0], repos[0], integration, {
+        stackRoot: 'new/root',
+        sourceRoot: 'source/root',
```
Comment (priscilawebdev) [inline tests/js/spec/views/organizationIntegrations/integrationCodeMappings.spec.jsx:139]: I'd replace these values reusing the constants defined above.

File: tests/js/spec/views/organizationIntegrations/integrationCodeMappings.spec.jsx (hunk 3 above)
Comment (priscilawebdev) [inline tests/js/spec/views/organizationIntegrations/integrationCodeMappings.spec.jsx:92]:
```suggestion
  });
  
  it('create new config', async () => {
```

File: tests/js/spec/views/organizationIntegrations/integrationCodeMappings.spec.jsx [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+        [pathConfig1, pathConfig2],
+      ],
+      [`/organizations/${org.slug}/repos/`, repos],
+    ]);
+
+    wrapper = mountWithTheme(
+      <IntegrationCodeMappings organization={org} integration={integration} />
+    );
+  });
+  it('shows the paths', () => {
```
Comment (priscilawebdev) [inline tests/js/spec/views/organizationIntegrations/integrationCodeMappings.spec.jsx:81]:
```suggestion
  });
  
  it('shows the paths', () => {
```

File: tests/js/spec/views/organizationIntegrations/integrationCodeMappings.spec.jsx [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+  });
+
+  it('opens modal', async () => {
+    expect(wrapper.find('input[name="stackRoot"]')).toHaveLength(0);
+    wrapper.find('button[aria-label="Add Mapping"]').first().simulate('click');
+    await tick();
+    expect(wrapper.find('input[name="stackRoot"]')).toHaveLength(1);
+  });
+
+  it('create new config', async () => {
```
Comment (priscilawebdev) [inline tests/js/spec/views/organizationIntegrations/integrationCodeMappings.spec.jsx:94]: you are not using `await` in this function, so the async can be removed

File: tests/js/spec/views/organizationIntegrations/integrationCodeMappings.spec.jsx [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+          repositoryId: repos[1].id,
+          stackRoot,
+          sourceRoot,
+          defaultBranch,
+        },
+      })
+    );
+  });
+
+  it('edit existing config', async () => {
```
Comment (priscilawebdev) [inline tests/js/spec/views/organizationIntegrations/integrationCodeMappings.spec.jsx:138]: here too

**Review by priscilawebdev — APPROVED** [review]

**Review by markstory — APPROVED** [review]

**Pull request merged**